### PR TITLE
[1322] Added send field on course serializer

### PR DESCRIPTION
**Pull request by defong**
### Context
Send field

### Changes proposed in this pull request
Added bool to denote is a course a Sen/Send course

### Guidance to review
retained the naming of `Sen` vs `Send` this is partly due to original naming as well as `Send` is an acronym and an actual word.

subject_code is hardcoded as it was from 
https://github.com/DFE-Digital/manage-courses-api/blob/e5c4f2a6ebe3644e322c4ff71d65ce7e83b3e403/src/ManageCourses.Domain/Models/Course.cs#L37

### CODE REVIEW

File: app/serializers/api/v2/serializable_course.rb [hunk 1]
```diff
@@ -16,6 +16,10 @@ class SerializableCourse < JSONAPI::Serializable::Resource
         SubjectMapper.get_subject_list(@object.name, ucas_subjects)
       end
 
+      attribute :is_sen do
```
Comment (dankmitchell) [inline app/serializers/api/v2/serializable_course.rb]: I think this should stay as `is_send` because `SEN` isn't used on its own anymore with the `D`

Comment (fofr) [reply app/serializers/api/v2/serializable_course.rb]: or `:is_send?`

Comment (defong) [reply app/serializers/api/v2/serializable_course.rb]: `is_send?` it will be then

File: spec/requests/api/v2/providers/courses_spec.rb [hunk 2]
```diff
@@ -11,14 +11,15 @@
 
   let(:course_subject_primary) { create(:subject, subject_name: 'Primary', subject_code: 'P') }
   let(:course_subject_mathematics) { create(:subject, subject_name: 'Mathematics', subject_code: 'M') }
+  let(:course_subject_sen) { create(:special_education_needs_subject) }
```
Comment (dankmitchell) [inline spec/requests/api/v2/providers/courses_spec.rb]: Id also update this to `course_subject_send`

File: spec/factories/subjects.rb [hunk 3]
```diff
@@ -15,5 +15,10 @@
     factory :futher_education_subject do
       subject_name { 'Further Education' }
     end
+
+    factory :special_education_needs_subject do
```
Comment (dankmitchell) [inline spec/factories/subjects.rb]: Shorten this to `send_subject` ?

**Comment by defong** [comment]
[Course - Basic detail tab SEND field (backend)](https://trello.com/c/HX65H45w/1322-course-basic-detail-tab-send-field-backend)

File: spec/serializers/api/v2/serializable_course_spec.rb [hunk 4]
```diff
@@ -79,5 +79,15 @@
 
       it { expect(subject["attributes"]).to include("funding" => "salary") }
     end
+
+    context "#is_send?" do
```
Comment (misaka) [inline spec/serializers/api/v2/serializable_course_spec.rb]: itym `describe` not `context` here.

File: spec/factories/subjects.rb [hunk 5]
```diff
@@ -15,5 +15,10 @@
     factory :futher_education_subject do
       subject_name { 'Further Education' }
     end
+
+    factory :send_subject do
```
Comment (misaka) [inline spec/factories/subjects.rb]: We seem to be tending towards using traits in the other factories, does this make more sense as a factory?

File: spec/factory_specs/subject_spec.rb [hunk 6]
```diff
@@ -1,12 +1,24 @@
...
+  describe "Subject 'Further Education' Factory" do
+    let(:subject) { create(:futher_education_subject) }
 
-describe "Subject 'Further Education' Factory" do
-  let(:subject) { create(:futher_education_subject) }
+    it "created subject" do
+      expect(subject).to be_instance_of(Subject)
+      expect(subject).to be_valid
+      expect(subject.subject_name).to eq 'Further Education'
+      expect(subject.in?(Subject.further_education)).to be true
```
Comment (misaka) [inline spec/factory_specs/subject_spec.rb]: How about using one-liners here?

```
let(:subject) { create(:further_education_subject) }

it { should be_instance_of(Subject) }
it { should be_valid }
its(:subject_name) { should eq 'Further Education' }
it { should be_in(Subject.further_education) }
```

If not, the last expect here could be this:

```
expect(subject).to be_in(Subject.further_education)
```

File: spec/factories/subjects.rb (hunk 5 above)
Comment (defong) [reply spec/factories/subjects.rb]: settled on making a factory for it as it related to actual record

File: spec/factory_specs/subject_spec.rb [hunk 7]
```diff
@@ -1,12 +1,24 @@
...
-    expect(subject).to be_instance_of(Subject)
-    expect(subject).to be_valid
-    expect(subject.subject_name).to eq 'Further Education'
-    expect(subject.in?(Subject.further_education)).to be true
+    it "created subject" do
+      expect(subject).to be_instance_of(Subject)
+      expect(subject).to be_valid
+      expect(subject.subject_name).to eq 'Special Educational Needs'
+      expect(subject.subject_code).to eq 'U3'
+    end
```
Comment (misaka) [inline spec/factory_specs/subject_spec.rb]: Same comment re: one-liners.

File: spec/factory_specs/subject_spec.rb (hunk 6 above)
Comment (defong) [reply spec/factory_specs/subject_spec.rb]: out of scope, but yeah

File: spec/factory_specs/subject_spec.rb (hunk 7 above)
Comment (defong) [reply spec/factory_specs/subject_spec.rb]: does that mean you going to let me keep factory

File: spec/serializers/api/v2/serializable_course_spec.rb (hunk 4 above)
Comment (defong) [reply spec/serializers/api/v2/serializable_course_spec.rb]: :sleepy:

File: spec/factory_specs/subject_spec.rb (hunk 6 above)
Comment (dankmitchell) [reply spec/factory_specs/subject_spec.rb]: I'll sort it separately as there's a typo in further anyway

Comment (dankmitchell) [reply spec/factory_specs/subject_spec.rb]: Done

**Comment by tvararu** [comment]
Merging this to progress the courses page. :shipit:
